### PR TITLE
Add on-device reset for half

### PR DESCRIPTION
**Pull request by jluceno**
Reduces host to device transfers to speed up test time. Perform output buffer resets on device rather on host. Observed a speed up of 1.5x over original.

### CODE REVIEW

File: test_conformance/half/Test_vStoreHalf.cpp [hunk 1]
```diff
@@ -783,18 +817,15 @@ int Test_vStoreHalf_private( cl_device_id device, f2h referenceFunc, d2h doubleR
...
                 dchk.aspace = addressSpaceNames[addressSpace];
 
-                cl_uint pattern = 0xdeaddead;
-                memset_pattern4( gOut_half, &pattern, BUFFER_SIZE/2);
-
-                error = clEnqueueWriteBuffer(gQueue, gOutBuffer_half, CL_FALSE,
-                                             0, count * sizeof(cl_half),
-                                             gOut_half, 0, NULL, NULL);
+                error = RunKernel(device, resetKernel, gInBuffer_single, gOutBuffer_half,
+                                  count, 0);
```
Comment (bashbaug) [inline test_conformance/half/Test_vStoreHalf.cpp:821]: Do you see the same benefits using [clEnqueueFillBuffer](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clEnqueueFillBuffer) on your device or is a kernel required?

**Review by bashbaug — COMMENTED** [review]
Is it possible to share which device is showing a 1.5x performance improvement and which command line you are running?  I am seeing a slight improvement with this change but not a 1.5x improvement.  Thanks!

File: test_conformance/half/Test_vStoreHalf.cpp (hunk 1 above)
Comment (jluceno) [reply test_conformance/half/Test_vStoreHalf.cpp:821]: I have not tested with clEnqueueFillBuffer. I can explore using this as opposed to a kernel.

**Comment by jluceno** [comment]
> Is it possible to share which device is showing a 1.5x performance improvement and which command line you are running? I am seeing a slight improvement with this change but not a 1.5x improvement. Thanks!

NVIDIA GeForce RTX 3080 (68CUs, 1800MHz)

I am running using cmd. The command is: `test_half_para.exe vstore_half`.

**Comment by bashbaug** [comment]
Checking: has this PR been supserseded by https://github.com/KhronosGroup/OpenCL-CTS/pull/1561?

**Comment by bashbaug** [comment]
I'm going to close this PR now that #1561 is merged.

**Pull request closed without merging**